relay/relay.go

```diff
@@ -22,44 +23,40 @@ type Relay struct {
 	broadcastClients            []*broadcastclient.BroadcastClient
 	broadcaster                 *broadcaster.Broadcaster
 	confirmedSequenceNumberChan chan arbutil.MessageIndex
-	messageChan                 chan broadcastFeedMessage
-}
-
-type broadcastFeedMessage struct {
-	message        arbstate.MessageWithMetadata
-	sequenceNumber arbutil.MessageIndex
+	messageChan                 chan broadcaster.BroadcastFeedMessage
 }
 
 type RelayMessageQueue struct {
-	queue chan broadcastFeedMessage
+	queue chan broadcaster.BroadcastFeedMessage
 }
 
-func (q *RelayMessageQueue) AddBroadcastMessages(pos arbutil.MessageIndex, messages []arbstate.MessageWithMetadata) error {
-	for i, message := range messages {
-		q.queue <- broadcastFeedMessage{
-			sequenceNumber: pos + arbutil.MessageIndex(i),
-			message:        message,
-		}
+func (q *RelayMessageQueue) AddBroadcastMessages(feedMessages []*broadcaster.BroadcastFeedMessage) error {
+	for _, feedMessage := range feedMessages {
+		q.queue <- *feedMessage
 	}
 
 	return nil
 }
 
-func NewRelay(serverConf wsbroadcastserver.BroadcasterConfig, clientConf broadcastclient.BroadcastClientConfig, chainId uint64, feedErrChan chan error) *Relay {
+func NewRelay(serverConf wsbroadcastserver.BroadcasterConfig, clientConf broadcastclient.Config, chainId uint64, feedErrChan chan error) *Relay {
 	var broadcastClients []*broadcastclient.BroadcastClient
 
-	q := RelayMessageQueue{make(chan broadcastFeedMessage, 100)}
+	q := RelayMessageQueue{make(chan broadcaster.BroadcastFeedMessage, 100)}
 
 	confirmedSequenceNumberListener := make(chan arbutil.MessageIndex, 10)
 
 	for _, address := range clientConf.URLs {
-		client := broadcastclient.NewBroadcastClient(address, chainId, 0, clientConf.Timeout, &q, feedErrChan)
+		client := broadcastclient.NewBroadcastClient(clientConf, address, chainId, 0, &q, feedErrChan, nil)
 		client.ConfirmedSequenceNumberListener = confirmedSequenceNumberListener
 		broadcastClients = append(broadcastClients, client)
 	}
 
+	dataSignerErr := func([]byte) ([]byte, error) {
+		log.Crit("relay attempted to sign feed message")
+		return nil, errors.New("relay attempted to sign feed message")
```

Comment:
I think `log.Crit` exits the node, so we should either use `log.Error` or maybe we can just get rid of the error return? If we keep the `log.Crit` we should add a comment noting that it exits.